### PR TITLE
[issue #1084] 解耦AuthServer，将功能拆解到UserOperator及StrategyOperator

### PR DESCRIPTION
**Pull request by KarKLi**
**Please provide issue(s) of this PR:**
Implement #1084

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] ApiServer
- [X] Auth
- [ ] Configuration
- [ ] Naming
- [ ] HealthCheck
- [ ] Metrics
- [ ] Docs
- [ ] Installation
- [ ] Performance and Scalability
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

Remove AuthServer and split it into UserOperator and StrategyOperator.

Change notes:

- Remove AuthServer
- Migrate GetAuthChecker&AfterResourceOperation method into StrategyOperator
- Migrate Login method into UserOperator
- Migrate GroupOperator into UserOperator, so it can be a part of UserOperator (group is a set of user)

- [X] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: auth/api.go [hunk 1]
```diff
@@ -73,6 +46,11 @@ type AuthChecker interface {
 
 // UserOperator 用户数据管理 server
 type UserOperator interface {
```
Comment (chuntaojun) [inline auth/api.go:48]: UserOperator => UserServer

File: auth/api.go [hunk 2]
```diff
@@ -128,6 +111,11 @@ type GroupOperator interface {
 
 // StrategyOperator 策略相关操作
 type StrategyOperator interface {
```
Comment (chuntaojun) [inline auth/api.go:113]: StrategyServer

File: auth/auth.go [hunk 3]
```diff
@@ -29,35 +29,80 @@ import (
...
+func RegisterStrategyOperator(s StrategyOperator) error {
 	name := s.Name()
-	if _, ok := Slots[name]; ok {
-		return errors.New("auth manager name is exist")
+	if _, ok := StrategyMgnSlots[name]; ok {
+		return errors.New("user manager name is exist")
 	}
 
-	Slots[name] = s
+	StrategyMgnSlots[name] = s
```
Comment (chuntaojun) [inline auth/auth.go:96]: StrategyMgnSlots = strategyMgnSlots

File: auth/defaultauth/strategy_authability.go [hunk 4]
```diff
@@ -24,13 +24,49 @@ import (
...
+	"github.com/polarismesh/polaris/store"
 )
 
+type startegyAuthAbility struct {
+	authMgn *defaultAuthChecker
+	target  *server
+}
+
+// Initialize 执行初始化动作
+func (svr *startegyAuthAbility) Initialize(authOpt *auth.Config, storage store.Store,
```
Comment (chuntaojun) [inline auth/defaultauth/strategy_authability.go:41]: startegyAuthAbility => strategyAuthAbility

File: auth/defaultauth/strategy_test.go [hunk 5]
```diff
@@ -103,17 +103,23 @@ func newStrategyTest(t *testing.T) *StrategyTest {
...
+		User: auth.UserConfig{
+			Name:   "",
+			Option: map[string]interface{}{},
+		},
+		Strategy: auth.StrategyConfig{
+			Name: "",
+			Option: map[string]interface{}{
+				"consoleOpen": true,
+				"clientOpen":  true,
+				"salt":        "polarismesh@2021",
```
Comment (chuntaojun) [inline auth/defaultauth/strategy_test.go:115]: salt 配置应该是 User 的配置，Strategy 对于 Token 的解析交给 UserServer 进行处理

**Review by chuntaojun — CHANGES_REQUESTED** [review]

File: auth/auth.go [hunk 6]
```diff
@@ -77,20 +123,37 @@ func Initialize(ctx context.Context, authOpt *Config, storage store.Store, cache
 
 // initialize 包裹了初始化函数，在 Initialize 的时候会在自动调用，全局初始化一次
 func initialize(_ context.Context, authOpt *Config, storage store.Store, cacheMgn *cache.CacheManager) error {
-	name := authOpt.Name
+	name := authOpt.User.Name
```
[the remark this answers is not in the document]
Comment (KarKLi) [reply auth/auth.go:126]: 在defaultauth中的auth_checker_test.go添加了单测。

File: release/standalone/docker-compose/server/polaris-server.yaml [hunk 7]
```diff
@@ -258,16 +258,20 @@ apiservers:
...
+  # auth的option现已迁移至user及strategy
+  # 在auth.option配置依然有效，但是会收到警告，提醒您尽快迁移配置至user及strategy级别下
+  user:
+    name: defaultUserManager
+  strategy:
+    name: defaultStrategyManager
+    option:
+      # token 加密的 salt，鉴权解析 token 时需要依靠这个 salt 去解密 token 的信息
+      # salt 的长度需要满足以下任意一个：len(salt) in [16, 24, 32]
+      salt: polarismesh@2021
```
[the remark this answers is not in the document]
Comment (KarKLi) [reply release/standalone/docker-compose/server/polaris-server.yaml:268]: 已统一修改，下同

**Comment by chuntaojun** [comment]
需要修复下CI

File: auth/defaultauth/auth_checker.go [hunk 8]
```diff
@@ -50,17 +50,50 @@ type defaultAuthChecker struct {
 
 // Initialize 执行初始化动作
 func (d *defaultAuthChecker) Initialize(options *auth.Config, s store.Store, cacheMgn *cache.CacheManager) error {
-	contentBytes, err := json.Marshal(options.Option)
-	if err != nil {
-		return err
-	}
+	// 新版本鉴权策略配置均从auth.Option中迁移至auth.user.option及auth.strategy.option中
+	var strategyContentBytes []byte
```
[the remark this answers is not in the document]
Comment (KarKLi) [reply auth/defaultauth/auth_checker.go]: done

**Pull request merged**